DIPs/dip10ss.md

```diff
@@ -0,0 +1,100 @@
+# Enum and Function Parameter Attributes
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | 10ss                                                            |
+| Review Count:   | NA                                                              |
+| Author(s):      | @skl131313                                                      |
+| Status:         | NA                                                              |
+
+
+## Abstract
+
+Allow additional meta information (attributes) to be attached with enums and functions parameters, including built-in attributes such as `deprecated`.
+
+### Links
+
+[Does it require a DIP?](http://forum.dlang.org/thread/cltyrthdxkkfvuxqasqw@forum.dlang.org)
+
+## Rationale
+
+It is currently not possible to attach attributes to both enums and function parameters. This excludes a few features that can be used with almost any other symbol in D. Attributes and User Defined Attributes serve as a means to provide extra meta data for a symbol. What can be said for why attributes were included as a feature in D can also be said for why they should be extended to enums and function parameters. It is benefitial to provide extra meta data about a symbol that can be used at compile-time.
```

Comment:
"user-defined" is the official spelling, see https://github.com/dlang/dlang.org/pull/2003#discussion_r157333463